spec/controllers/spree/api/shipments_controller_spec.rb

```diff
@@ -0,0 +1,113 @@
+require 'spec_helper'
+
+describe Spree::Api::ShipmentsController, type: :controller do
+  render_views
+
+  let!(:shipment) { create(:shipment) }
+  let!(:attributes) { [:id, :tracking, :number, :cost, :shipped_at, :stock_location_name, :order_id, :shipping_rates, :shipping_method, :inventory_units] }
+
+  before do
+    allow(controller).to receive(:spree_current_user) { current_api_user }
+  end
+
+  context "as an admin" do
+    let!(:order) { shipment.order }
+    let(:order_ship_address) { create(:address) }
+    let!(:stock_location) { create(:stock_location_with_items) }
+    let!(:variant) { create(:variant) }
+    let(:params) do
+      { quantity: 2,
+        variant_id: stock_location.stock_items.first.variant.to_param,
+        order_id: order.number,
+        stock_location_id: stock_location.to_param,
+        format: :json }
+    end
+
+    before do
+      order.update_attribute(:ship_address_id, order_ship_address.id)
+    end
+
+    sign_in_as_admin!
+
+    context '#create' do
+      it 'creates a shipment if order does not have a shipment' do
+        order.shipment.destroy
+        order.reload
+
+        spree_post :create, params
+
+        validate_response
+        expect(json_response["inventory_units"].size).to eq 2
+        expect(order.reload.line_items.first.variant.price).to eq(variant.price)
+      end
```

Comment:
and what if  `@order.shipments.create` fails?

Reply:
yeah, I wondered. spree code is not doing anything and not testing either...
I tested existing spree code and the new adaptation.

What do you think I should do?
1- change the code to rescue some exception and add error codes? we would have to handle errors in shipments.create, @order.contents.add and @shipment.save!
2- just cover an error case with a spec? create a spec that makes shipments.create raise and then assert it is raising?

Reply:
The answer to your question is
if shipments.create fails, the endpoint will return a http error code. for the user that means adding the first variant to a new order will fail.
option 2 above would be to cover this http error code with a spec.

Reply:
ok, so that means it is somehow dealt with and it doesn't blow into pieces. IMO we should cover so with specs. We'll see then if we like what is returned and it needs to be worked on in an upcoming PR.

Reply:
ok, I'll do that :+1:

Reply:
ok, I'll do that.

Reply:
done, pls have a look @sauloperez